File: src/types/lsp.ts
/**
 * LSP (Language Server Protocol) 相关类型定义
 */

/** LSP 服务器状态 */
export type LSPServerStatus = 'not_installed' | 'installing' | 'installed' | 'starting' | 'running' | 'error';

/** LSP 服务器类型 */
export type LSPServerType = 'typescript' | 'javascript' | 'rust' | 'python';

/** LSP 服务器配置 */
export interface LSPServerConfig {
  /** 服务器类型 */
  type: LSPServerType;
  /** 是否启用 */
  enabled: boolean;
  /** 当前状态 */
  status: LSPServerStatus;
  /** 安装方式 */
  installMethod: 'bundled' | 'system' | 'manual';
  /** 版本号 */
  version?: string;
  /** 可执行路径 */
  path?: string;
  /** 错误信息 */
  error?: string;
}

/** LSP 配置 */
export interface LSPConfig {
  /** 是否启用 LSP 功能 */
  enabled: boolean;
  /** 各语言服务器配置 */
  servers: Record<LSPServerType, LSPServerConfig>;
}

/** LSP 位置信息 */
export interface LSPLocation {
  /** 文件 URI */
  uri: string;
  /** 范围 */
  range: {
    start: { line: number; character: number };
    end: { line: number; character: number };
  };
}

/** LSP 补全项 */
export interface LSPCompletionItem {
  /** 标签（显示文本） */
  label: string;
  /** 类型 */
  kind: number;
  /** 详情 */
  detail?: string;
  /** 文档 */
  documentation?: string;
  /** 插入文本 */
  insertText: string;
}

/** LSP 诊断信息 */
export interface LSPDiagnostic {
  /** 范围 */
  range: {
    start: { line: number; character: number };
    end: { line: number; character: number };
  };
  /** 严重程度：1=Error, 2=Warning, 3=Information, 4=Hint */
  severity: 1 | 2 | 3 | 4;
  /** 消息 */
  message: string;
  /** 来源 */
  source?: string;
}

/** 默认 LSP 配置 */
export const defaultLSPConfig: LSPConfig = {
  enabled: true,
  servers: {
    typescript: {
      type: 'typescript',
      enabled: true,
      status: 'installed', // WASM 版本始终已安装
      installMethod: 'bundled',
    },
    javascript: {
      type: 'javascript',
      enabled: true,
      status: 'installed', // WASM 版本始终已安装
      installMethod: 'bundled',
    },
    rust: {
      type: 'rust',
      enabled: false,
      status: 'not_installed',
      installMethod: 'system',
    },
    python: {
      type: 'python',
      enabled: false,
      status: 'not_installed',
      installMethod: 'system',
    },
  },
};

File: src-tauri/src/models/lsp.rs
/**
 * LSP 相关模型定义
 */

use serde::{Deserialize, Serialize};

/// LSP 服务器状态
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum LSPServerStatus {
    NotInstalled,
    Installing,
    Installed,
    Starting,
    Running,
    Error,
}

/// LSP 服务器类型
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
#[serde(rename_all = "lowercase")]
pub enum LSPServerType {
    TypeScript,
    JavaScript,
    Rust,
    Python,
}

/// LSP 服务器信息
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LSPServerInfo {
    /// 服务器类型
    pub language: LSPServerType,
    /// 是否启用
    pub enabled: bool,
    /// 当前状态
    pub status: LSPServerStatus,
    /// 版本号
    pub version: Option<String>,
    /// 可执行路径
    pub path: Option<String>,
    /// 错误信息
    pub error: Option<String>,
}

/// LSP 位置信息
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LSPLocation {
    /// 文件 URI
    pub uri: String,
    /// 范围
    pub range: LSPRange,
}

/// LSP 范围
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LSPRange {
    pub start: LSPPosition,
    pub end: LSPPosition,
}

/// LSP 位置
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LSPPosition {
    pub line: u32,
    pub character: u32,
}

/// LSP 补全项
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LSPCompletionItem {
    /// 标签
    pub label: String,
    /// 类型
    pub kind: u32,
    /// 详情
    pub detail: Option<String>,
    /// 文档
    pub documentation: Option<String>,
    /// 插入文本
    pub insert_text: Option<String>,
}

/// LSP 诊断信息
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LSPDiagnostic {
    /// 范围
    pub range: LSPRange,
    /// 严重程度
    pub severity: u32,
    /// 消息
    pub message: String,
    /// 来源
    pub source: Option<String>,
}

/// LSP 悬停信息
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LSPHover {
    /// 内容
    pub contents: String,
}

/// LSP 检查结果
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LSPCheckResult {
    pub status: LSPServerStatus,
    pub path: Option<String>,
    pub version: Option<String>,
    pub error: Option<String>,
}

/// LSP 安装结果
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LSPInstallResult {
    pub success: bool,
    pub error: Option<String>,
}
